deepinterpolation/inferrence_collection.py

```diff
@@ -154,124 +261,190 @@ def __init__(self, inferrence_json_path, generator_obj):
+    def _create_h5_datasets(self,
+                            output_dataset_name: str,
+                            raw_dataset_name: str,
+                            ):
+        """Create datasets on output h5 file. 
+        """
         if self.output_padding:
-            first_sample = self.generator_obj.start_sample - \
-                self.generator_obj.start_frame
-            final_shape = [self.nb_datasets*self.batch_size+first_sample]
+            final_shape = [self.nb_datasets*self.batch_size + self.first_sample]
 
         else:
             final_shape = [self.nb_datasets*self.batch_size]
-            first_sample = 0
 
         final_shape.extend(self.indiv_shape[:-1])
-
         chunk_size = [1]
         chunk_size.extend(self.indiv_shape[:-1])
-
+    
         with h5py.File(self.output_file, "w") as file_handle:
-            dset_out = file_handle.create_dataset(
-                "data",
+            file_handle.create_dataset(
+                output_dataset_name,
                 shape=tuple(final_shape),
                 chunks=tuple(chunk_size),
                 dtype=self.output_datatype,
             )
+            if self.save_raw:
+                file_handle.create_dataset(
+                    raw_dataset_name,
+                    shape=tuple(final_shape),
+                    chunks=tuple(chunk_size),
+                    dtype=self.output_datatype,
+                )
+        logger.info(f"Created empty HDF5 file {self.output_file}")
 
-            # Initialize onset of output sample movie
-            local_start = first_sample
-
-            for epoch_index, index_dataset in enumerate(tqdm(np.arange(0, self.nb_datasets, self.steps_per_epoch))):
 
-                local_length = np.min([self.steps_per_epoch, self.nb_datasets-index_dataset])
-                
-                # We overwrite epoch_index to allow the last unfilled epoch
-                self.generator_obj.epoch_index = epoch_index
+    def _write_output_to_file(self,
+                              start: int,
+                              end: int,
+                              corrected_data: np.ndarray,
+                              corrected_raw: np.ndarray = None,
+                              ):
+        """Write denoised and raw outputs onto respective datasets on
+        h5 file
+        """
+        with h5py.File(self.output_file, "a") as file_handle:
+            dset_out = file_handle[self.output_dataset_name]
+            dset_out[start:end] = np.squeeze(corrected_data, -1)
 
-                predictions_data = self.model.predict(
-                        self.generator_obj,
-                        steps = local_length, max_queue_size = 10,
-                        workers = self.workers,
-                        use_multiprocessing = self.use_multiprocessing)
+            if self.save_raw:
+                raw_out = file_handle[self.raw_dataset_name]
+                raw_out[start:end] = np.squeeze(corrected_raw, -1)
```

Comment:
need to check whether `corrected_raw` is `None` before doing `np.squeeze`, otherwise will crash